### PR TITLE
PhpUnitStrictFixer - extend risky note

### LINKED ISSUE #3746: php_unit_strict should not use assertSame on values that might be objects
`php_unit_strict` is marked risky because "risky when any of the functions are overridden"

In reality, it exchanges all instances of `assertEquals` to `assertSame`, which breaks tests with objects that are equal but not identical. Likewise `assertNotEquals` and `assertNotSame` will fail too.

I doubt it's reasonable to statically analyse the tests to determine whether its parameters are objects, but if either side is a literal it should be doable. If this is a wontfix, please update the readme to warn about it.

### PR DESCRIPTION
**Pull request by jnvsor**
Closes #3746

### CODE REVIEW

File: src/Fixer/PhpUnit/PhpUnitStrictFixer.php [hunk 1]
```diff
@@ -74,7 +74,7 @@ public function testSomeTest()
                 ),
             ],
             null,
-            'Risky when any of the functions are overridden.'
+            'Risky when any of the functions are overridden, or you are testing object equality.'
```
Comment (SpacePossum) [inline src/Fixer/PhpUnit/PhpUnitStrictFixer.php]: Thanks for the fast and nice proposal!
I would drop the Oxford Comma and rephrase to;
 `'Risky when any of the functions are overridden or when testing object equality.'`
just to stay in the same style of the rest of most of the README, we kind of like to have the same style with everything here, not stating one is better than the other ^_^

Comment (jnvsor) [reply src/Fixer/PhpUnit/PhpUnitStrictFixer.php]: Fixed

> we kind of like to have the same style with everything here

That's kind of the whole point of `php-cs-fixer` right? :smile: 

How can I tell which is the lowest maintained branch?

Comment (julienfalque) [reply src/Fixer/PhpUnit/PhpUnitStrictFixer.php]: The lowest maintained branch is 2.2 currently. I think we don't have any documentation about that.

**Comment by keradus** [comment]
Thank you @jnvsor.

**Pull request merged**